src/t.jl

```diff
@@ -28,18 +28,18 @@ export OneSampleTTest, TwoSampleTTest, EqualVarianceTTest,
 @compat abstract type TTest <: HypothesisTest end
 @compat abstract type TwoSampleTTest <: TTest end
 
-pvalue(x::TTest; tail=:both) = pvalue(TDist(x.df), x.t; tail=tail)
-
-default_tail(test::TTest) = :both
+pvalue(x::TTest; tail=x.tail) = pvalue(TDist(x.df), x.t; tail=tail)
+tail(x::TTest)  = x.tail    # defaults set by constructors
```

Comment:
In another branch, I already started to add the `alpha` field to all types, and the `tail` field those types for which the `pvalue` method has a tail parameter. I also develop corresponding unit tests and documentation.

Apparently, these changes are quite extensive. We should discuss what the present PR should be going for.

Initially, this PR was meant to (only) provide the _infrastructure_ for subtypes of HypothesisTest to have `tail` and `alpha` fields. Initially, I was not planning to actually add these fields to all types. Starting to do so confirmed my impression that this should be a different work package.

I see that all tests _should_ have `alpha` and `tail` fields (if applicable), but I suggest to delay these changes to a follow-up PR. I'd like the present PR to focus on the infrastructure, as initially intended. Of course, I will be applying the changes required to make this PR a complete extension to the HypothesisTests package.